src/main/scala/edu/ie3/simona/model/participant/HpModel.scala

```diff
@@ -222,24 +230,32 @@ final case class HpModel(
       relevantData: HpRelevantData,
       isRunning: Boolean,
       demandWrapper: ThermalDemandWrapper,
+      currentThermalGridState: ThermalGridState,
   ): HpState = {
-    val lastStateStorageQDot = lastState.thermalGridState.storageState
+    val qDotLastHouseState = lastState.thermalGridState.houseState
       .map(_.qDot)
       .getOrElse(zeroKW)
+    val currentEnergyOfThermalStorage = currentThermalGridState.storageState
+      .map(_.storedEnergy)
+      .getOrElse(zeroKWh)
+    val pThermalOfStorage = thermalGrid.heatStorage
+      .map(_.getpThermalMax: squants.Power)
+      .getOrElse(zeroKW)
 
     val (newActivePowerHp, newThermalPowerHp, qDotIntoGrid) = {
       if (isRunning)
         (pRated, pThermal, pThermal)
-      else if (lastStateStorageQDot < zeroKW)
-        (zeroKW, zeroKW, lastStateStorageQDot * (-1))
+      // If the house has req. demand and storage isn't empty, we can heat the house from storage.
       else if (
-        lastStateStorageQDot == zeroKW && (demandWrapper.houseDemand.hasRequiredDemand || demandWrapper.heatStorageDemand.hasRequiredDemand)
+        currentEnergyOfThermalStorage > zeroKWh && demandWrapper.houseDemand.hasRequiredDemand
+      ) {
+        (zeroKW, zeroKW, pThermalOfStorage)
+        // Edge case when em controlled: If the house was heated last state by Hp and surplus energy is gone now,
+        // but house didn't reach target temperature yet. House can be heated from storage, if this one is not empty.
+      } else if (
+        currentEnergyOfThermalStorage > zeroKWh && demandWrapper.houseDemand.hasAdditionalDemand && qDotLastHouseState > zeroKW
```

Comment:
Whats additionalDemand for the house again and why doesn't requiredDemand suffice, since the comment line above mentions target temperature, and the documentation of thermal demand says that "boundary or target temperature" is reached when the _required_ demand is fulfilled?

Reply:
Good catch. At this point, additionalDemand is still when "boundary or target temperature" are reached, depending on the setPower the Em demands. But this will change to target temperature with #1183